index/scorch/segment/zap/chunk.go

```diff
@@ -0,0 +1,40 @@
+package zap
+
+import (
+	"fmt"
+)
+
+// LegacyChunkMode was the original chunk mode (always chunk size 1024)
+// this mode is still used for chunking doc values.
+var LegacyChunkMode uint32 = 1024
+
+// DefaultChunkMode is the most recent improvement to chunking and should
+// be used by default.
+var DefaultChunkMode uint32 = 1025
+
+func getChunkSize(chunkMode uint32, cardinality uint64, maxDocs uint64) (uint64, error) {
+	switch {
+	// any chunkMode <= 1024 will always chunk with chunkSize=chunkMode
+	case chunkMode <= 1024:
+		// legacy chunk size
+		return uint64(chunkMode), nil
```

Comment:
curious to know - why chunkMode(default modes too) is a uint32 when we need a uint64 back?

Reply:
In the file format (footer) we only use 32 bits to record chunk mode.  So, chunk mode really is only 32-bits, but i think the IntCoder takes chunkMode as uint64, probably because we do math with it against other uint64s.  So, basically someone has to do some converting somewhere.  I no longer know what is best.